### DISCUSSION TITLE
Fighting CORS in React App and Airflow with Docker-Compose

### DISCUSSION THREAD

**Discussion by MashiPe** [apache/airflow#29033] [Q&A] [answered]
So I'm building a web app with React  and want to execute a DAG through the Ariflow API, read the documentation about CORS [here](https://airflow.apache.org/docs/apache-airflow/stable/security/api.html#enabling-cors), added the values in my airflow.cfg file:

```
access_control_allow_headers = origin, content-type, accept
access_control_allow_methods = POST, GET, OPTIONS, DELETE
access_control_allow_origins = *
```

and keep getting the following CORS error:

![Screenshot from 2023-01-19 03-44-23](https://user-images.githubusercontent.com/101341198/213395585-818d23c6-37ee-444f-88c9-224c4950e8b5.png)

I'm doing the request with axios:

```          
var body_2 = { dag_run_id: uuidv4() }

await axios.post(url,JSON.stringify(body_2),{
    headers:{
        'Content-Type':'application/json',
    },
    auth:{
        username:'airflow',
        password:'airflow'
    },
})
```

Also, i'm aware that i can use de web interface from airflow but I need to run de DAG from my app. What am i missing?

**Comment by Taragolis** [comment, marked as answer]
You need to specify correct headers in your request if you want to use CORS: see https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Reply by MashiPe** [reply]
Looks like I'm just an idiot and didn't realise that i was using the wrong URL, the airflow configuration and the request headers are fine, thanks.
